src/utils/resolve.ts

```diff
@@ -162,33 +165,55 @@ function fullResolve(
       fileExistsCache.set(cacheKey, resolved.path as string | null)
       return resolved
     }
-  } else {
-    const configResolvers = settings['import-x/resolver-legacy'] ||
-      settings['import-x/resolver'] || {
-        node: settings['import-x/resolve'],
-      } // backward compatibility
-
-    for (const { enable, options, resolver } of normalizeConfigResolvers(
-      configResolvers,
+  } else if (
+    // backward compatibility for very old `import-x/resolve` options that is no longer supported
+    Object.prototype.hasOwnProperty.call(settings, 'import-x/resolve') &&
```

Comment:
Why `hasOwnProperty` check needed?

Reply:
This is to align with some previous code. Probably not needed, but ESLint settings can be any non-plain objects now anyway (with `eslint.config.js`).

Reply:
Any thing special should be considered vs `settings['import-x/resolve']`?

Reply:
```js
// eslint.config.js
export default [{
  settings: new Proxy({ get })
}]
```

Since eslint flat config allows any valid JS, it doesn't hurt to have a `hasOwnProperty` check anyway.

Reply:
Doesn't that an intended behavior from the user? I still don't quite follow the benefits to check `hasOwnProperty`.

Reply:
Again, this is to align with some previous code. I can remove both of checks.

Reply:
Done in [`28189c5` (#272)](https://github.com/un-ts/eslint-plugin-import-x/pull/272/commits/28189c590ea66b764166a0de2ac753928444e85e)

Reply:
Thanks, sorry for my nitpick.